src/nvim/ops.c

```diff
@@ -3085,8 +3085,10 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
       memset(ptr, ' ', (size_t)bd.endspaces);
       ptr += bd.endspaces;
       // move the text after the cursor to the end of the line.
-      memmove(ptr, oldp + bd.textcol + delcount,
-              (size_t)((int)oldlen - bd.textcol - delcount + 1));
+      size_t columns;
+      STRICT_SUB(oldlen, bd.textcol, &columns, size_t);
+      STRICT_SUB(columns, delcount +1, &columns, size_t);
```

Comment:
This entire change is not at all readable. Can't we have something like `STRICT_CAST((int)oldlen - bd.textcol - delcount + 1, size_t)` (first assert the int is >= 0, then cast to size_t).

Reply:
> This entire change is not at all readable. Can't we have something like `STRICT_CAST((int)oldlen - bd.textcol - delcount + 1, size_t)` (first assert the int is >= 0, then cast to size_t).

As long as we know the result fits in `int`, sure

Reply:
columns and vcols can be assumed to fit in an `int`. (`int` intermediate calculations are also implicitly asserted by UBSAN for overflow).

Reply:
@justinmk @bfredl 
It looks like there is no consensus. What would you like me to do here?

Reply:
@bfredl how will STRICT_CAST avoid the PVS warning?

Reply:
I would suggest just write `int columns = (int)oldlen - bd.textcol - delcount + 1` and then `assert(columns >= 0)` for now, unless you are interested in constructing the new macro.

Reply:
@justinmk Can we know if PVS generates a new warning or not without writing the new code?

Reply:
So the `int columns = (int)oldlen - bd.textcol - delcount + 1` will remove PVS warning as there is no casting of a result anymore (only operands).
On one hand I agree with @bfredl - this code is more readable. I've seen emacs codebase, it is totally unreadable because of macros. On the other hand `STRICT_ADD/SUB` feels like a safer way.

I don't know how strong are the guarantees that `oldlen` fits into `int`

In the vacuum the problem with suggested solution is that this:
`int columns = (int)oldlen - bd.textcol - delcount + 1` +  `assert(columns >= 0)` 
can pass assertion in case of double overflow, if numbers are big enough. Not sure how real this is in given codebase.

Reply:
> I don't know how strong are the guarantees that oldlen fits into int

`oldlen` is a column number (number of columns of the line). Nvim restricts column numbers to values that fits into a signed `int`.

> In the vacuum the problem with suggested solution is that this:
int columns = (int)oldlen - bd.textcol - delcount + 1 + assert(columns >= 0)
can pass assertion in case of double overflow

It will not pass with UBSAN enabled, which already checks for overflow after every signed operation.

Reply:
Ok, I have nothing to object as I have no idea how UBSAN works.

Reply:
done